src/contexts/providers/AuthProvider.js

```diff
@@ -0,0 +1,21 @@
+import { useContext, createContext, useReducer } from "react";
+import authReducer from "../reducers/authReducer";
+
+const AuthContext = createContext(null);
+
+const AuthProvider = ({ children }) => {
+  const [authState, authDispatch] = useReducer(authReducer, {
+    user: {},
+    isAuthenticated: localStorage.getItem("token") ? true : false,
```

Comment:
if you've a token, `isAuthenticated `should not be set to be true because your token may be expired.

Reply:
I will keep this in mind when I transition from Mockbee to own backend.